crates/next-core/src/page_source.rs

```diff
@@ -97,7 +95,9 @@ pub async fn create_page_source(
         src_pages
     } else {
         return Ok(NoContentSourceVc::new().into());
-    };
+    }
+    .resolve()
+    .await?;
 
     let client_ty = Value::new(ClientContextType::Pages { pages_dir });
```

Comment:
the `pages_dir` in `client_ty` was not matching the `pages_dir` in the client_ty created within the client transition, as it wasn't resolved (still pointing to the `join` operation).

Note in future we should prevent unresolved Vcs to be stored in structs without explict opt-in of that.